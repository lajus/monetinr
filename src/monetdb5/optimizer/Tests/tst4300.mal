#some join operations

#mdb.start();
b:= bat.new(:oid,:lng);
barrier v:= 0:lng;
        bat.append(b,v);
        redo v:= iterator.next(1:lng,100:lng);
exit v;

i:= aggr.count(b);
io.print(i);

c:= algebra.select(b,0:lng,9:lng);
i:= aggr.count(c);
io.print(i);
d:= algebra.select(b,0:lng,49:lng);
i:= aggr.count(d);
io.print(i);

bo := batcalc.oid(b);
co := batcalc.oid(c);
cor:= bat.reverse(co);
zz:= algebra.joinPath(bo,cor,d);
i:= aggr.count(zz);
io.print(i);
