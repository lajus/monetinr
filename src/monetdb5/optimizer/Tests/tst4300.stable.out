stdout of test 'tst4300` in directory 'monetdb5/optimizer` itself:


# 23:52:52 >  
# 23:52:52 >  Mtimeout -timeout 60 Mserver "--config=/ufs/mk/monet5/src/mal/Tests/All.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5/Linux/lib/MonetDB5:/ufs/mk/opensource/MonetDB/Linux/lib/MonetDB" --set "gdk_dbfarm=/ufs/mk/opensource/MonetDB/Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/opensource/MonetDB/Linux/var/MonetDB5/log" --set mapi_port=39352 --set sql_port=45171 --set xquery_port=52900 --set monet_prompt= --trace --dbname=mTests_src_optimizer  tst4300.mal </dev/null
# 23:52:52 >  

# MonetDB server v5.9.0, based on kernel v1.27.0
# Serving database 'mTests_src_optimizer', using 4 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008- MonetDB B.V., all rights reserved
# Visit http://monetdb.cwi.nl/ for further information
# Listening for connection requests on mapi:monetdb://eir.ins.cwi.nl:32348/
function user.main():void;
#some join operations 
#mdb.start(); 
    b := bat.new(:oid,:lng);
barrier v := 0:lng;
    bat.append(b,v);
    redo v := iterator.next(1:lng,100:lng);
exit v;
    i := aggr.count(b);
    io.print(i);
    c := algebra.select(b,0:lng,9:lng);
    i := aggr.count(c);
    io.print(i);
    d := algebra.select(b,0:lng,49:lng);
    i := aggr.count(d);
    io.print(i);
    bo := batcalc.oid(b);
    co := batcalc.oid(c);
    cor := bat.reverse(co);
    zz := algebra.joinPath(bo,cor,d);
    i := aggr.count(zz);
    io.print(i);
#to be detected by join inversion 
#z:= joinPath(b,c); 
#zz:= joinPath(zz,d); 
#i:= aggr.count(z); 
#io.print(i); 
end main;
[ 100 ]
[ 10 ]
[ 50 ]
[ 10 ]

# 23:52:52 >  
# 23:52:52 >  Done.
# 23:52:52 >  

